crates/optimism/rpc/src/eth/transaction.rs

```diff
@@ -77,16 +77,18 @@ where
     N: FullNodeComponents,
 {
     type Transaction = Transaction;
+    type TransactionError = OpEthApiError;
 
     fn fill(
         &self,
         tx: TransactionSignedEcRecovered,
         tx_info: TransactionInfo,
-    ) -> Self::Transaction {
+    ) -> Result<Self::Transaction, Self::TransactionError> {
         let signed_tx = tx.clone().into_signed();
         let hash = tx.hash;
 
-        let mut inner = EthTxBuilder.fill(tx, tx_info);
+        let mut inner =
+            EthTxBuilder.fill(tx, tx_info).expect("EthTxBuilder.fill should be infallible");
```

Comment:
only this call to fill can be expected to be infallible. all the other expect statements need to be removed.